pephub/routers/eido/eido.py

```diff
@@ -377,28 +113,57 @@ async def validate(
 
             p = peppy.Project(f"{dirpath}/{init_file.filename}")
 
-    # save schema string to temp file, then read in with eido
-    with tempfile.NamedTemporaryFile(mode="w") as schema_file:
-        schema_file.write(schema)
-        schema_file.flush()
+    if schema is None and schema_registry is None:
+        raise HTTPException(
+            status_code=400,
+            detail={
+                "error": "Must supply either a registry path or a list of files to validate."
+            },
+        )
+    
+    if schema_registry is not None:
+        schema_url = f"https://schema.databio.org/{schema_registry}.yaml"
+            
         try:
-            schema_dict = eido.read_schema(schema_file.name)[0]
-        except eido.exceptions.EidoSchemaInvalidError as e:
+            response = requests.get(schema_url)
+            response.raise_for_status()  # Check if the request was successful
+            yaml_string = response.text
+        except requests.exceptions.RequestException as e:
             raise HTTPException(
-                status_code=200,
-                detail={"error": f"Schema is invalid: {str(e)}"},
+                status_code=400,
+                detail={"error": f"Error fetching schema: {str(e)}"},
             )
 
-    # validate
+        # save schema string to temp file
+        with tempfile.NamedTemporaryFile(mode='w', delete=False) as yaml_file:
+            yaml_file.write(yaml_string)
+            schema_dict = yaml_file.name
+    else:
+        # save schema string to temp file, then read in with eido
+        with tempfile.NamedTemporaryFile(mode="w") as schema_file:
+            schema_file.write(schema)
+            schema_file.flush()
+            try:
+                schema_dict = eido.read_schema(schema_file.name)[0]
+            except eido.exceptions.EidoSchemaInvalidError as e:
+                raise HTTPException(
+                    status_code=200,
+                    detail={"error": f"Schema is invalid: {str(e)}"},
```

Comment:
Is this different than above when we return `400` for non-existent schema?

Reply:
It's not, that one slipped, was intending to update that as well to 400